packages/web/app/components/Appointments/LocationBookingsTable.jsx

```diff
@@ -299,12 +299,15 @@ export const LocationBookingsTable = ({ patient }) => {
   });
 
   // Query to check if there are past location bookings
-  const pastBookingsQuery = useLocationBookingsQuery({
-    patientId: patient?.id,
-    before: getCurrentDateTimeString(),
-    after: '1970-01-01 00:00',
-    rowsPerPage: 1,
-  });
+  const pastBookingsQuery = useLocationBookingsQuery(
+    {
+      patientId: patient?.id,
+      before: getCurrentDateTimeString(),
+      after: '1970-01-01 00:00',
+      rowsPerPage: 1,
+    },
+    { keepPreviousData: true },
```

Comment:
The query uses all the fetch options as query key. Given before is different every time, it always refetch and creates the glitch. I've added `keepPreviousData` to keep displaying the previous query's data while a new query is loading, in order to fix the glitch.

Reply:
In the other places it's used, are we calculating the `before` live, so the same issue could occur?

Reply:
Yep I did have a look, there are live calculated `before` and `after`, but they do not get updated as frequently as here. An example is:
```
before: values.startTime ? toDateTimeString(endOfDay(parseISO(values.startTime))) : null
```

which gets the `values.startTime` from a date picker, so I thought it's acceptable to leave it as it is in order to avoid regression risk

Reply:
Yes that's great - I didn't word that very well but by live I was meaning `now()` type of thing